### PR TITLE
build: JSON schema extension and validation

### LINKED ISSUE #5: RFC: Add (optional?) entry for mod GitHub repo to `mods.json`
Having the GitHub repo linked in there would help with automations that compare Thunderstore package with GitHub repo to check for modifications etc

This would change

```json
{
    "Fifty.mp_frostbite": {
        "DependencyPrefix": "Fifty-Frostbite",
        "Versions" : [
            { 
                "Version": "0.0.1",
                "Checksum": "8cf111c9ac2ab1521677769702e10be4db8d47c0eca16bda197aceb895cf2b5c"
            }
        ]
    }
}
```
to

```json
{
    "Fifty.mp_frostbite": {
        "DependencyPrefix": "Fifty-Frostbite",
        "Repo": "https://github.com/some-user/some-repo",
        "Versions" : [
            { 
                "Version": "0.0.1",
                "Checksum": "8cf111c9ac2ab1521677769702e10be4db8d47c0eca16bda197aceb895cf2b5c"
            }
        ]
    }
}
```

I'm not sure if we should make the key optional or not as maps might have an issue in general with being hosted on GitHub due to their file sizes.

### PR DESCRIPTION
**Pull request by Alystrasz**
**META PR, NO MOD VERIFICATION NEEDED**

This PR introduces a JSON schema, `.github/schema.json`, that allows checking the format of the `verified-mods.json` manifesto file.
A GitHub action allows automatic verification of the manifesto format.

New fields are required:
* each mod entry must feature a `"Repository"` entry;
* each mod version entry must feature a `"CommitHash"` entry.

To reflect those changes, I added `mp_s2s` as a verified mod example, since it matches our verified mods criteria (as opposed to `mp_frostbite` that isn't public at the time of writing).

### GitHub Action runs:

**Manifesto with wrong format:** https://github.com/Alystrasz/VerifiedMods/actions/runs/7010991283/job/19072700693
**Manifesto with good format:** https://github.com/Alystrasz/VerifiedMods/actions/runs/7011669230/job/19074720803

Closes #5, #6, #7 and #10.

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  verify-json-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Validate package.json against local schema
+        uses: cardinalby/schema-validator-action@v3
+        with:
+            file: 'verified-mods.json'
+            schema: '.github/schema.json'
```
Comment (GeckoEidechse) [inline .github/workflows/validate.yml:14]: Insert trailing newline? 👉👈

File: .github/schema.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "^[A-Za-z._0-9]+": {
+            "type": "object",
+            "properties": {
+                "DependencyPrefix": {
+                    "type": "string",
+                    "pattern": "^([A-Za-z._0-9]+)-([A-Za-z._0-9]+)$"
+                },
+                "Repository": {
+                    "type": "string",
+                    "pattern": "^https://"
```
Comment (GeckoEidechse) [inline .github/schema.json:16]: Should we even require GitHub repo or also allow other platforms? ^^

**Review by GeckoEidechse — COMMENTED** [review]
Awesome work :D

This PR is like the "mega kill" equivalent of closing GitHub issues, haha

File: .github/schema.json (hunk 2 above)
Comment (Alystrasz) [reply .github/schema.json:16]: I mean, I have nothing against GitLab, that's why I only forced the `https` prefix

File: .github/workflows/validate.yml (hunk 1 above)
Comment (Alystrasz) [reply .github/workflows/validate.yml]: Addressed in 51c12c32ed20f85bfe55da93ee12d16215155f07

File: .github/schema.json (hunk 2 above)
Comment (GeckoEidechse) [reply .github/schema.json:16]: Works for me :D

**Comment by GeckoEidechse** [comment]
This PR is so nicely formatted, I'm just gonna merge the whole thing as one <3

**Pull request merged**